src/helpers/deploy.ts

```diff
@@ -9,35 +12,47 @@ interface DeployResult {
   address: string
 }
 
-export function deploy(url: string, privateKey: string, bin: string): Promise<DeployResult> {
+export function deploy(url: string, privateKey: string, binPath: string): Promise<DeployResult> {
+  const transactionConfirmationBlocks = 3
+  const options = {
+    transactionConfirmationBlocks,
+  }
+
   const web3 = new Web3(new Web3.providers.HttpProvider(url))
   privateKey = add0x(privateKey)
 
   const { address } = web3.eth.accounts.wallet.add(privateKey)
 
-  const data = add0x(fs.readFileSync(bin).toString())
+  const data = add0x(fs.readFileSync(binPath).toString())
 
-  const contract = new web3.eth.Contract([])
+  const Contract: any = web3.eth.Contract // ts hack: transactionConfirmationBlocks is not a valid option
+  const contract = new Contract([], undefined, options)
 
   const deploy = contract.deploy({ data })
+  let receipt: TransactionReceipt
 
   return deploy
     .estimateGas({
       from: address,
     })
-    .then(gas => {
-      const contract = deploy.send({
-        from: address,
-        gas,
-      })
-
-      const receiptPromise = new Promise<TransactionReceipt>(resolve => {
-        contract.on('receipt', receipt => resolve(receipt))
-      })
-
-      return Promise.all([contract, receiptPromise])
-    })
-    .then(([contract, receipt]) => {
-      return { address: contract.options.address, receipt }
+    .then((gas: Unit) => {
+      cli.log(`Estimated gas: ${gas}`)
```

Comment:
Maybe these logs should be sent to stderr, so that the JSON in the output can be piped. `cli-ux` doesn't seem to have a way to do that, so `console.error` should do for now. Maybe later we can add a `--verbose` flag to all commands.

Reply:
Done, I replaced it by `process.stderr.write` from NodeJs.